src/BuildScriptGenerator/DotNetCore/DefaultAspNetCoreWebAppProjectFileProvider.cs

```diff
@@ -132,19 +133,42 @@ public string GetProjectFile(ISourceRepo sourceRepo)
         // To enable unit testing
         internal static bool IsAspNetCoreWebApplicationProject(XDocument projectFileDoc)
         {
-            var webSdkProjectElement = projectFileDoc.XPathSelectElement(
-                DotnetCoreConstants.WebSdkProjectXPathExpression);
-            return webSdkProjectElement != null;
+            // For reference
+            // https://docs.microsoft.com/en-us/visualstudio/msbuild/project-element-msbuild?view=vs-2019
+
+            // Look for the attribute value on Project element first as that is more common
+            // Example: <Project Sdk="Microsoft.NET.Sdk.Web">
+            var expectedWebSdkName = DotnetCoreConstants.WebSdkName.ToLowerInvariant();
```

Comment:
why not just use `OrdinalIgnoreCase` in both if statements and skip this `ToLowerInvariant()` call?